TS04-Dplyr-And-Time-Series-In-R.Rmd

```diff
@@ -198,134 +354,99 @@ manipulate the data in different ways:
+`R` to skip `NA` values.
 
-Let's calculate the mean air temperature for each Julian day. Since we have a
-few missing values we can add `na.rm=TRUE` to force R to ignore any NA values
-when making the calculations. 
 
 ```{r dplyr-summarize}
 harMet15.09.11 %>%
-  group_by(jd) %>%
+  group_by(year, jd) %>%
   summarize(mean_airt = mean(airt, na.rm = TRUE))  
 ```
 
-Julian days repeat 1-365 (or 366) each year, therefore what we have here is that
-the mean temperature for January 1st across 2009, 2010, and 2011 was -3.7
-Celsius. 
+<div id="challenge" markdown="1">
+##Challenge
 
-Sometimes we may want to summarize our data this way, other times we may want 
-to summarize our air temperature data for each day in each of the three years.  
-To do that we need to group our data by two different values at once, year and 
-Julian day. 
-
-###Extract Year from a Date-Time Column
-Currently, our date and time information is in one column, `datetime`, but to 
-group by year, we're first going to need to create a year-only column. For this 
-we'll use the `lubridate` package. Since our date column is already a POSIXct 
-date-time class, we can use the `year()` function. 
+We can use `sum` to calculate the total rather than mean value for each Julian
+Day. Using this information, do the following:
 
-```{r dplyr-lubridate-2}
-harMet15.09.11$year <- year(harMet15.09.11$datetime)
-```
+1. Calculate total `prec` for each Julian Day over the 3 years - name your
+data.frame `total.prec`. 
+2. Create a plot of Daily Total Precipitation for 2009-2011. 
+3. Add a title and x and y axis labels.
+4. If you use `qplot` to create your plot, use `colour=as.factor(total.prec$year)`
+to color the data points by year.
 
-Using `names()` we can see that we now have a `year` variable . 
-```{r dplyr-lubridate-3}
-#check to make sure it worked
-names(harMet15.09.11)
-str(harMet15.09.11$year)
-```
+</div>
 
-Now we have our two variables: `jd` and `year`. To get the mean air temperature
-for each day for each year we can use the `dplyr` pipes to group by year and
-Julian day.
+```{r challenge-answer, echo=FALSE }
 
-```{r dplyr-2-groups}
-harMet15.09.11 %>%
+total.prec <- harMet15.09.11 %>%
   group_by(year, jd) %>%
-  summarize(mean_airt = mean(airt, na.rm = TRUE))
+  summarize(sum_prec = sum(prec, na.rm = TRUE)) 
+
+
+qplot(total.prec$jd, total.prec$sum_prec,
+      main="Total Precipitation",
+      xlab="Julian Day", ylab=="Precip (mm)", 
+      colour=as.factor(total.prec$year))
+
 ```
 
-Given just the header in the output we can see the difference between the
-3-year average temperature for 1 January (-3.7C, previous analysis) and the 
-average temperature for 1 January 2009 (-15.1C, this analysis).
 
-###Use Mutate & Combining Functions to Increase Efficiency
-To create more efficient code, could we create the year variable within our
-`dplyr` function call? 
+###Mutate - Add Data.Frame Columns 2 Dplyr Output
```

Comment:
Mutate - Add Data.Frame Colums To dplyr Output    

all othere titles leave package names uncapitalized.